doc: add entgql edge ordering (#3482)
ent/ent: doc/md/tutorial-todo-gql-paginate.md

By adding this annotation to the `Todo` schema, the `orderBy` argument will be changed from `TodoOrder` to `[TodoOrder!]`.

## Order By Edge Count

> The `OrderField` annotation can be attached to non-unique edges to enable sorting nodes based on the count of specific
> edge types.

Reviewer: ```suggestion
Non-unique edges may be annotated with The `OrderField` annotation to enable sorting nodes based on the count of specific
edge types.
```

```go title="ent/schema/todo/go"
func (Todo) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("children", Todo.Type).
			Annotations(
				entgql.RelayConnection(),
				// highlight-next-line
				entgql.OrderField("CHILDREN_COUNT"),
			).
			From("parent").
			Unique(),
	}
}
```

:::info
The naming convention for this ordering term is: `UPPER(<edge-name>)_COUNT`. For example, `CHILDREN_COUNT`
or `POSTS_COUNT`.
:::

## Order By Edge Field

> The `OrderField` annotation can be attached to unique edges, allowing nodes to be sorted by their associated edge fields.
> For example, _sorting posts by their author's name_, or _ordering todos based on their parent's priority_. Note that

Reviewer: ```suggestion
Unique edges can be annotated with the `OrderField` annotation to allow nodes to be sorted by their associated edge fields.
For example, _sorting posts by their author's name_, or _ordering todos based on their parent's priority_. Note that
```
